contracts/beyondFramework/dca/base/DCABaseUpgradeable.sol

```diff
@@ -0,0 +1,1148 @@
+                ) * _bluechipTokenScale()) /
+                currentBluechipBalance;
+        }
+
+        // set proper strategy state
+        setBluechipInvestmentState(BluechipInvestmentState.EmergencyExited);
+
+        emit StatusChanged(
+            BluechipInvestmentState.Investing,
+            BluechipInvestmentState.EmergencyExited
+        );
+    }
+
+    function _emergencyWithdrawUserDeposit(address sender) private {
+        uint256 notInvestedYet;
+        uint256 investedIntoBluechip;
+
+        DCADepositor storage depositor = depositors[sender];
+        for (uint256 i = 0; i < depositor.positions.length; i++) {
+            (
+                uint256 positionBluechipInvestment,
+                uint256 positionNotInvestedYet
+            ) = _computePositionWithdrawAll(depositor.positions[i]);
+
+            investedIntoBluechip += positionBluechipInvestment;
+            notInvestedYet += positionNotInvestedYet;
+        }
+
+        // since depositor withdraws everything
+        // we can remove his data completely
+        delete depositors[sender];
+
+        // if deposit token != emergency exit token compute share
+        if (depositTokenInfo.token != emergencyExitDepositToken.token) {
+            uint256 convertedDepositShare = (notInvestedYet *
+                emergencySellDepositPrice) / depositTokenScale;
+
+            if (convertedDepositShare != 0) {
+                emergencyExitDepositToken.token.safeTransfer(
+                    sender,
+                    convertedDepositShare
+                );
+            }
+        } else {
+            // otherwise send deposit token
+            if (notInvestedYet != 0) {
+                depositTokenInfo.token.safeTransfer(sender, notInvestedYet);
+            }
+        }
+
+        // if bluechip != emergency exit token compute share
+        if (_bluechipAddress() != address(emergencyExitBluechipToken.token)) {
+            uint256 convertedBluechipShare = (investedIntoBluechip *
+                emergencySellBluechipPrice) / _bluechipTokenScale();
+
+            if (convertedBluechipShare != 0) {
+                emergencyExitBluechipToken.token.safeTransfer(
+                    sender,
+                    convertedBluechipShare
+                );
+            }
+        } else {
+            // otherwise send bluechip token
+            if (investedIntoBluechip != 0) {
+                _transferBluechip(sender, investedIntoBluechip);
+            }
+        }
+    }
+
+    // ----- Base Class Setters -----
+    function setBluechipInvestmentState(BluechipInvestmentState newState)
+        public
+        onlyOwner
+    {
+        bluechipInvestmentState = newState;
+    }
+
+    function setDepositFee(DepositFee memory newDepositFee) public onlyOwner {
+        require(
+            newDepositFee.feeReceiver != address(0),
+            "Invalid fee receiver"
+        );
+        require(newDepositFee.fee <= 100, "Invalid fee percentage");
```

Comment:
100 should be changed to 10000 accordingly.

`require(newDepositFee.fee <= 10000, "Invalid fee percentage");`